CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 # Note: update this to your new project's name and version
 project(
   Greeter
-  VERSION 1.0
+  VERSION 1.1
```

Comment:
We are not updating the starter project version in CMake as it's only meant to be an example. Actual starter versions can be seen in the [releases](https://github.com/TheLartians/ModernCppStarter/releases) of this repo.

Reply:
OK, but I run into the problems, as the old version headers were installed and TEST_INSTALLED_VERSION failed!

Reply:
Note that the intended way to test the installation locally is not to pollute the system, but to create a separate root dir:

```bash
# install the library
cmake -S. -Bbuild/install -DCMAKE_PREFIX_PATH=$(pwd)/root -DCMAKE_INSTALL_PREFIX=$(pwd)/root
cmake --build build/install --target install
# test the library
cmake -Stest -Bbuild/test -DCMAKE_PREFIX_PATH=$(pwd)/root -DTEST_INSTALLED_VERSION=1
cmake --build build/test
cmake --build build/test --target test
```

Reply:
OK, and a little more:
```shel
# all together
cmake -S all -B build/all -G "${GENERATOR}" -DCMAKE_PREFIX_PATH=${CURDIR}/root -DTEST_INSTALLED_VERSION=1 -DENABLE_TEST_COVERAGE=1
cmake --build build/all
cmake --build build/all --target test
cmake --build build/all --target check-format

# check the library
cmake -S standalone -B build/standalone -G "${GENERATOR}" -DCMAKE_PREFIX_PATH=${CURDIR}/root -DCMAKE_EXPORT_COMPILE_COMMANDS=1
cmake --build build/standalone --target all
run-clang-tidy.py -p build/standalone -checks='-*,modernize-*' standalone
```

Reply:
Ok, I think I see a major misconception: the purpose of the `all` directory is to join all target's for simplified development. It's not meant to be used by library end-users or installers. The project should only ever be used or installed from its root directory. That way also no test flags or other issues will get leaked into the library.

Reply:
but it is  OK to use **`all`** as developer in an **IDE**!

On the other site, **`all`** looks not good with` visual code` or `clion`, I do not know why.

With **qtcretor**, it works fine.